lib/report.js

```diff
@@ -201,11 +201,10 @@ const tapTestCaption = (id, test) => {
 class TapReporter extends Reporter {
   constructor(options) {
     super(options);
-    if (!this.options.type) {
-      this.options.type = this.options.stream.isTTY ? 'classic' : 'tap';
+    if (this.options.type && this.options.type !== 'tap') {
```

Comment:
What's wrong with passing the `'tap'` type to `TSR` constructor?

Reply:
The `TapReporter` already generates tap output, therefore there is no need to put it through TSR.

Also, since TSR doesn't support being `require`d in the `worker_threads` therefore this also allows us to use `metatests` from `worker_threads`.

Reply:
I thought that `TSR` at least validates the syntax when using it with `'tap'` as a constructor argument.